Dependency missing in docker build instructions

<!--
If you are reporting a new issue, make sure that we do not have any duplicates
already open. You can ensure this by searching the issue list for this
repository. If there is a duplicate, please close your issue and add a comment
to the existing issue instead.
-->

**Description**

Running `make` results in the following:
```
# github.com/containerd/containerd/vendor/github.com/containerd/btrfs
vendor/github.com/containerd/btrfs/btrfs.go:38:10: fatal error: btrfs/ioctl.h: No such file or directory
 #include <btrfs/ioctl.h>
          ^~~~~~~~~~~~~~~
compilation terminated.
make: *** [Makefile:188: bin/containerd] Error 2
```

It appears that btrfs/ioctl.h is in libbtrfs-dev instead of btrfs-tools. Change the Dockerfile in BUILDING.md in the "Build containerd and runc" section to include libbtrfs-dev instead of btrfs-tools.

**Steps to reproduce the issue:**
1. Follow the `Via Docker container` -> `Build containerd and runc` instructions in BUILDING.md
2. Run `make`

**Describe the results you received:**

See above description, build does not complete successfully.

**Describe the results you expected:**

Build to complete successfully.

**Output of `containerd --version`:**

N/A

**Any other relevant information:**

This issue is related to https://github.com/containerd/containerd/issues/3813, but unfortunately not all of the documentation in BUILDING.md was updated.

--- a/BUILDING.md
+++ b/BUILDING.md
@@ -174,7 +174,7 @@ We can build an image from this `Dockerfile`:
 FROM golang
 
 RUN apt-get update && \
-    apt-get install -y btrfs-tools libseccomp-dev
+    apt-get install -y libbtrfs-dev libseccomp-dev
 
 ```
 

